internal/runpartition/delimiter_substitution.go

```diff
@@ -0,0 +1,51 @@
+package runpartition
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/rwx-research/captain-cli/internal/errors"
+	"github.com/rwx-research/captain-cli/internal/templating"
+)
+
+type DelimiterSubstitution struct {
```

Comment:
great as it is, but the file and the code makes me think this should be `PartitioningTemplate` and include the template in the model. This would make the `ValidateTemplate` and `Example` functions make more sense in context

Reply:
I think that could make sense. I was taking inspiration from the existing retry substitutions. Perhaps we can take another pass when adding framework specific partition subs?